src/test/fastfilter_tests.cpp

```diff
@@ -37,45 +37,51 @@ BOOST_AUTO_TEST_CASE(fastfilter_tests)
             BOOST_CHECK(filt.contains(tmp));
             BOOST_CHECK(!filt.checkAndSet(tmp));
         }
-        BOOST_CHECK(collisions < 400); // sanity check, actual result may vary
+        BOOST_CHECK(collisions < 10); // sanity check, actual result may vary
```

Comment:
The following python code makes the assumption that each test is independent.  This is not quite true since a prior collision means that fewer bits will be actually set in the data, resulting in a slightly smaller probability of subsequent collision.

```import math

def bloomCollision(n,m, k):
    return (1.0 - math.e**(-1*float(k)*float(n)/float(m)))**k

def Test():
    probs= []
    for i in range(50000):
        probs.append(bloomCollision(i,1000000, 16))

    s = sum(probs)
    m = 1.0
    for i in probs:
        m *= i
    oneCollision = s-m
    tenCollision = oneCollision**10
    print(tenCollision)```

So looks like 6.430521732653953e-06

Reply:
Ok, that's good enough!